plonky2x/src/frontend/eth/mpt/builder.rs

```diff
@@ -51,7 +51,26 @@ impl<L: PlonkParameters<D>, const D: usize> CircuitBuilder<L, D> {
             _phantom: PhantomData,
         };
         self.add_simple_generator(generator.clone());
-        generator.output
+
+        const NUM_LIMBS: usize = 64;
+        // FIXME: overflow? potentially
+        // might have to use a zipped iterator and compute lt
+        let max = self.constant(L::Field::from_canonical_u64(1 << (NUM_LIMBS - 1)));
+        let _one: Variable = self.constant(L::Field::from_canonical_u64(1));
+        // from circomlib: https://github.com/iden3/circomlib/blob/master/circuits/comparators.circom#L89
+        let tmp = self.add(lhs, max);
+        let res = self.sub(tmp, rhs);
+        // typically stored in BE, but we can just index instead of reversing
+        let res_bits = self.api.split_le(res.0, NUM_LIMBS);
+        let last_bit = Variable::from(res_bits[NUM_LIMBS - 1].target);
+        let lt_var = self.sub(_one, last_bit);
+        let lt = BoolVariable::from(lt_var);
+        let eq = self.is_equal(lhs, rhs);
+        let lte = self.or(lt, eq);
+
+        let output = generator.output;
+        self.assert_is_equal(lte, output);
```

Comment:
In this case, we should just return `lte` and remove the generator

Reply:
fixed in https://github.com/succinctlabs/succinctx/pull/152/commits/105ccb5087670c4af0c5749a92639e84e88c73d9